tests/src/test/java/org/eclipse/hono/tests/http/TelemetryHttpIT.java

```diff
@@ -214,6 +221,10 @@ public void testUploadQos1MessageFailsIfDeliveryStateNotUpdated(final VertxTestC
 
         httpResponseFuture
                 .onComplete(ctx.succeeding(response -> {
+                    ctx.verify(() -> {
+                        assertThat(response.bodyAsString())
+                                .isEqualTo(new SendMessageTimeoutException("").getClientFacingMessage());
+                    });
```

Comment:
wow, this looks terrible, doesn't it ;-)
Maybe we can add a *static String ServiceInvocationException.getLocalizedMessage(String)* method which accepts a key instead?

Reply:
Well, I agree that it doesn't look pretty and I was actually a bit hesitant, using that. However, it's in a test and as an advantage, the resource handling and keys can be kept internal to the exception classes that way.

But ok, I've changed it to use *ServiceInvocationException.getLocalizedMessage()* now.